src/Http/Controllers/Requests/DefaultFormRequest.php

```diff
@@ -20,6 +21,31 @@ protected function prepareForValidation(): void
         }
     }
 
+    /**
+     * @param Validator $validator
+     * @return void
+     */
+    protected function failedValidation(Validator $validator)
+    {
+        if (config('media-library.disk_name') !== 'local') {
+            $this->clearTempMediaFiles($this->json->all());
```

Comment:
What's this for?

Reply:
In the `prepareMedia` function we generate a temp file if you don't use the `local` disk. This will remain on your server, otherwise it won't go away. So with that function you make sure that your temp files are cleaned up nicely at the end of the validation.